### PR TITLE
Use evolution calendar data source

### LINKED ISSUE #2: Module 'gnome_next_meeting_applet.applet' has no attribute 'CREDENTIALS_PATH'
* Gnome Next Meeting applet version: 0.3.0-1~focal1
* Python version: 3.8.5
* Operating System: Ubuntu 20.04.2 TLS

### Description

Couldn't start the application

### What I Did
I tried to open using the credential as mentioned in the README file. But i get the following error:
```sh
$ gnome-next-meeting-applet-auth ~/Documents/next-meeting.json

Traceback (most recent call last):
  File "/usr/bin/gnome-next-meeting-applet-auth", line 11, in <module>
    load_entry_point('gnome-next-meeting-applet==0.1.0', 'console_scripts', 'gnome-next-meeting-applet-auth')()
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 490, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 2854, in load_entry_point
    return ep.load()
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 2445, in load
    return self.resolve()
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 2451, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/usr/lib/python3/dist-packages/gnome_next_meeting_applet/auth.py", line 24, in <module>
    import gnome_next_meeting_applet.applet as gnma
  File "/usr/lib/python3/dist-packages/gnome_next_meeting_applet/applet.py", line 32, in <module>
    from gi.repository import AppIndicator3 as appindicator
  File "/usr/lib/python3/dist-packages/gi/importer.py", line 132, in load_module
    raise ImportError('cannot import name %s, '
ImportError: cannot import name AppIndicator3, introspection typelib not found
```
then I installed the missing lib by `sudo apt install libappindicator3-dev`

After that I tried to run it again, but the following error is occurred:
```sh
$ gnome-next-meeting-applet-auth ~/Documents/next-meeting.json

/usr/lib/python3/dist-packages/gnome_next_meeting_applet/applet.py:32: PyGIWarning: AppIndicator3 was imported without specifying a version first. Use gi.require_version('AppIndicator3', '0.1') before import to ensure that the right version gets loaded.
  from gi.repository import AppIndicator3 as appindicator
Traceback (most recent call last):
  File "/usr/bin/gnome-next-meeting-applet-auth", line 11, in <module>
    load_entry_point('gnome-next-meeting-applet==0.1.0', 'console_scripts', 'gnome-next-meeting-applet-auth')()
  File "/usr/lib/python3/dist-packages/gnome_next_meeting_applet/auth.py", line 46, in main
    dpath = pathlib.Path(gnma.CREDENTIALS_PATH)
AttributeError: module 'gnome_next_meeting_applet.applet' has no attribute 'CREDENTIALS_PATH'
```

What am I missing?

### PR DESCRIPTION
**Pull request by chmouel**
Remove the whole auth thing and direct query to google calendar api
since it's too much of a PITA for users.

Use evolution calendar data sources instead.

Thanks to @GabLeRoux for the integration code.

Fixes #2, #5, #3 

Signed-off-by: Chmouel Boudjnah <chmouel@chmouel.com>

### CODE REVIEW

**Comment by chmouel** [comment]
Pushed some fixes where we were badly parsing the user tz and not fallback onto UTC (checked gnome calendar source code to see how they did it).

There is some doublons coming thru the wire too so making sure we won't parses it.

testing it a litle bit more and maybe think about packaging before release.

**Comment by GabLeRoux** [comment]
oh this is exciting! I will definitely give it a try! :D

File: README.md [hunk 1]
```diff
@@ -61,46 +60,13 @@ A configured example is located in the [config.sample.yaml](./config.samples.yam
...
 
-* You can now launch the gnome-next-meeting applet.
+All calendars are configured directly in Gnome Online Account setting, it will grab it from there.
 
 ### Starting
 
-You can just do : 
+You can just do :
 
 Alt+F2 and start `gnome-next-meeting-applet` make sure you have done the oauth
```
Comment (GabLeRoux) [inline README.md:71]: We don't need to dance anymore here :dancers:

**Comment by GabLeRoux** [comment]
There's also a few files in `.github/screenshot/` that can be removed (such as `create-oauth-1.png`)

File: README.md (hunk 1 above)
Comment (chmouel) [reply README.md]: You are right thanks ! 🎯

**Comment by chmouel** [comment]
I just tested on a brand new install with a blank VM and there was quite a lot of bugs when not running with my setup for the first time, i just fixed most of the bugs there and now running on fedora : 

fedorarunning

**Comment by chmouel** [comment]
Should be good to go, could not find any bugs today..

File: gnome_next_meeting_applet/evolution_calendars.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            uuid = value.get_id().get_uid()
+            # sometime we get doublons whatever the reasons is
+            if uuid in seen:
+                continue
+            seen.append(uuid)
+            events.append(value)
+        return events
+
+    # TODO: calendar filtering
+    def get_all_events(self, restrict_to_calendar=[]):
```
Comment (GabLeRoux) [inline gnome_next_meeting_applet/evolution_calendars.py:100]: Default argument value is mutable 

workaround:

```python
    def get_all_events(self, restrict_to_calendar=None):
        if restrict_to_calendar is None:
            restrict_to_calendar = []
```

Comment (chmouel) [reply gnome_next_meeting_applet/evolution_calendars.py:100]: oh you are totally right, I always forget this gotcha with python... I think pylint in a github actions will be able to catch this,

**Pull request merged**